vscode-trace-extension/src/trace-explorer/trace-tree.ts

```diff
@@ -164,7 +210,27 @@ export const fileHandler = (analysisTree: AnalysisProvider) => async (context: v
             }
         }
 
-    })();
+        if (token.isCancellationRequested) {
+            if (experiment) {
+                experimentManager.deleteExperiment(experiment.UUID);
+            }
+            rollbackTraces(traces, 20, progress);
+            progress.report({ message: ProgressMessages.COMPLETE, increment: 10 });
+            panel.dispose();
+            return;
+        }
+        progress.report({ message: ProgressMessages.COMPLETE, increment: 30 });
+    });
+};
+
+const rollbackTraces =async (traces: Array<TspTrace>, progressIncrement: number, progress: vscode.Progress<{
+    message?: string | undefined;
```

Comment:
`message:` would suffice, as `undefined` is already specified (both being mutually redundant here).